apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet/ErrorReporting/GoogleWebApiExceptionLogger.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright 2018 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Cloud.Diagnostics.Common;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http.Controllers;
+
+namespace Google.Cloud.Diagnostics.AspNet
+{
+    /// <summary>
+    /// Google Cloud Error Reporting Logger for Web Api applications.
+    /// Reports exceptions to Stackdriver Error Reporting API.
+    /// </summary>
+    /// 
+    /// <example>
+    /// <code>
+    /// string projectId = "[Google Cloud Platform project ID]";
+    /// string serviceName = "[Name of service]";
+    /// string version = "[Version of service]";
+    /// var exceptionLogger = GoogleWebApiExceptionLogger.Create(projectId, serviceName, version);
+    ///
+    /// try
+    /// {
+    ///     string scores = File.ReadAllText(@"C:\Scores.txt");
+    ///     Console.WriteLine(scores);
+    /// }
+    /// catch (IOException e)
+    /// {
+    ///     exceptionLogger.Log(e, ActionContext);
+    /// }
+    /// </code>
+    /// </example>
+    public class GoogleWebApiExceptionLogger : IWebApiExceptionLogger, IDisposable
```

Comment:
I don't have enough knowledge of Web API here - do we need this to be specific to Web API applications vs just "normal ASP.NET"? What's the difference here?

(Is it just that it uses HttpActionContext instead of other kinds of context?)

Reply:
Yes, see my comment [here](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/pull/2304#issuecomment-402990294). Unfortunately in Asp.Net Web API and MVC expose HTTP context info differently.